File: guanacos_api/app/models/Rol.js
'use strict';

module.exports = (sequelize, DataTypes) => {
  const Rol = sequelize.define('Rol',{
    rol: {
      type: DataTypes.STRING,
      allowNull: false
    },
    image: {
      type: DataTypes.STRING,
      allowNull: false
    },
    route: {
      type: DataTypes.STRING,
      allowNull: false
    }
  },{
    tableName: "roles",freezeTableName: true
  });

  Rol.associate = function(models) {
    Rol.belongsToMany(models.User, {
        as: "users", 
        through: "user_rol", 
        foreignKey: "idrol"
      });
  }
  
  return Rol;
};

File: guanacos_app/lib/src/models/Rol.dart
import 'user_rol.dart';

class Rol {
  int? id;
  String? rol;
  String? image;
  String? route;
  UserRol? userRol;

  Rol({
    this.id,
    this.rol,
    this.image,
    this.route,
    this.userRol,
  });

  factory Rol.fromJson(Map<String, dynamic> json) => Rol(
    id: json["id"],
    rol: json["rol"],
    image: json["image"],
    route: json["route"],
    userRol: UserRol.fromJson(json["user_rol"]),
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "rol": rol,
    "image": image,
    "route": route,
    "user_rol": userRol!.toJson(),
  };
}